lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -1401,6 +1402,188 @@ static void maybeDiagnoseClassWithoutInitializers(ClassDecl *classDecl) {
+      if (seen.insert(inherited).second) {
+        workList.push_back(inherited);
+      }
+    }
+
+    // Get the original conformance of the extended type to this protocol.
+    auto conformanceRef = TypeChecker::conformsToProtocol(
+        extendedType, protoDecl, ext);
+    if (!conformanceRef.isConcrete()) {
+      continue;
+    }
+    
+    // Check to see if this protocol is one we care about.
+    auto conformance = conformanceRef.getConcrete();
+    auto protoModule = protoDecl->getParentModule();
+
+    // If this protocol comes from the extension's module, skip it.
+    if (protoModule == module) {
+      continue;
+    }
+    
+    // If the protocol comes from the underlying clang module of the extension's
+    // module, skip it.
+    if (auto clangModule = protoModule->findUnderlyingClangModule()) {
+      if (clangModule->getTopLevelModuleName() == module->getName().str()) {
+        continue;
+      }
+    }
+    
+    // If that conformance came from this extension, then we warn. Otherwise
+    // we will have diagnosed it on the extension that actually declares this
+    // specific conformance.
+    if (conformance->getDeclContext() != ext) {
+      continue;
+    }
+    
+    // If we've come this far, we know this extension is the first declaration
+    // of the conformance of the extended type to this protocol.
+    externalProtocols.insert(protoDecl);
+  }
+
+  TypeRepr *extTypeRepr = ext->getExtendedTypeRepr();
+
+  // If we didn't find any violations, we're done.
+  if (externalProtocols.empty()) {
+    return;
+  }
+
+  llvm::SmallString<32> protocolList;
+
+  {
+    llvm::raw_svector_ostream os(protocolList);
+    llvm::interleaveComma(externalProtocols, os, [&os](ProtocolDecl *proto) {
+      os << "'" << proto->getName() << "'";
+    });
+  }
+  
+  ext->diagnose(
+      diag::extension_retroactive_conformance_in_resilient_module,
+      extendedNominalDecl->getName(),
+      externalProtocols.size() == 1,
+      protocolList.str());
+
+  auto diag = ext->diagnose(diag::extension_retroactive_conformance_silence);
+
+  // If the extended type name isn't qualified with the module name, add it.
+  if (!isModuleQualified(extTypeRepr, module)) {
+    llvm::SmallString<12> moduleNameAndDot;
+    llvm::raw_svector_ostream os(moduleNameAndDot);
+    os << extTypeModule->getName() << ".";
+    diag.fixItInsert(extTypeRepr->getStartLoc(), moduleNameAndDot.str());
+  }
+
+  for (TypeLoc tyLoc : ext->getInherited()) {
+    auto protoDecl = 
+        dyn_cast_or_null<ProtocolDecl>(tyLoc.getType()->getAnyNominal());
+    if (protoDecl && externalProtocols.remove(protoDecl)) {
+      llvm::SmallString<12> moduleNameAndDot;
+      llvm::raw_svector_ostream os(moduleNameAndDot);
+      os << protoDecl->getParentModule()->getName() << ".";
+      diag.fixItInsert(
+          tyLoc.getTypeRepr()->getStartLoc(), moduleNameAndDot.str());
+    }
+  }
+
+  if (externalProtocols.empty()) {
```

Comment:
This also occurs on lines [1529-1532](https://github.com/apple/swift/pull/36068/files#diff-fa1198ad7fbbed0b2ad70245eaa658121ebd138b3d92dcb8013d14ce906bfe58R1529-R1532)

Reply:
I will have removed an entry from this list on line 1562, but I'll add a comment mentioning that

Reply:
This code as-written is more clever than I'd like it to be